src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs

```diff
@@ -58,10 +59,14 @@ public partial class JsonTypeInfo
             {
                 // Set by JsonMetadataServices.
                 Debug.Assert(_elementTypeInfo == null);
+                Debug.Assert(value is not null);
                 _elementTypeInfo = value;
+                NumberHandlingIsApplicableToElementType = _elementTypeInfo.PropertyInfoForTypeInfo.NumberHandlingIsApplicable();
             }
         }
 
+        internal bool NumberHandlingIsApplicableToElementType { get; private set; }
```

Comment:
Nit: rather than storing this information here, wouldn't it make more sense to cache the result of `NumberHandlingIsApplicable()` in the original `JsonPropertyInfo`?